src/qt/rpcconsole.cpp

```diff
@@ -1031,7 +1031,10 @@ void RPCConsole::on_lineEdit_returnPressed()
     //: A console message indicating an entered command is currently being executed.
     message(CMD_REPLY, tr("Executing…"));
     m_is_executing = true;
-    Q_EMIT cmdRequest(cmd, m_last_wallet_model);
+
+    QMetaObject::invokeMethod(m_executor, [this, cmd]() {
+        m_executor->request(cmd, m_last_wallet_model);
```

Comment:
b7257271c533a97ea5e8b675ece6eee61aef23de

I think we should avoid `m_last_wallet_model` since this is executed asynchronously, and in theory `m_last_wallet_model` could change meanwhile. I suggest above:
```cpp
    WalletModel* wallet_model{nullptr};
#ifdef ENABLE_WALLET
    wallet_model = ui->WalletSelector->currentData().value<WalletModel*>();
```
and here:
```cpp
    QMetaObject::invokeMethod(m_executor, [=] {
        m_executor->request(cmd, wallet_model);
    });
```

Reply:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/589#issuecomment-1105104759).